File: generator/src/test/resources/swift/expected/RequestHeaderParamsTest/test-generation-of-multiple-header-parameters-with-inline-type-definitions.output.swift
import PotentCodables
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest(category: FetchTestCategoryHeaderParam, type: FetchTestTypeHeaderParam) async throws -> [String : AnyValue] {
    return try await self.requestFactory.result(
      method: .get,
      pathTemplate: "/tests",
      pathParameters: nil,
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: [
        "category": category,
        "type": type
      ]
    )
  }

  public enum FetchTestCategoryHeaderParam : String, CaseIterable, Codable {

    case politics = "politics"
    case science = "science"

  }

  public enum FetchTestTypeHeaderParam : String, CaseIterable, Codable {

    case all = "all"
    case limited = "limited"

  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestHeaderParamsTest/test-generation-of-multiple-header-parameters-with-inline-type-definitions.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import kotlin.Any
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public suspend fun fetchTest(category: FetchTestCategoryHeaderParam,
      type: FetchTestTypeHeaderParam): Map<String, Any> = this.requestFactory
    .result(
      method = Method.Get,
      pathTemplate = "/tests",
      acceptTypes = this.defaultAcceptTypes,
      headers = mapOf(
        "category" to category,
        "type" to type
      )
    )

  public enum class FetchTestCategoryHeaderParam {
    Politics,
    Science,
  }

  public enum class FetchTestTypeHeaderParam {
    All,
    Limited,
  }
}
